tools/critest/critest_diff.sh

```diff
@@ -0,0 +1,31 @@
+#! /bin/bash
+#
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You may
+# not use this file except in compliance with the License. A copy of the
+# License is located at
+#
+# 	http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is distributed
+# on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+# express or implied. See the License for the specific language governing
+# permissions and limitations under the License.
+
+
+FICD_DM_POOL="$FICD_DM_POOL"
```

Comment:
How about taking the input from the stdin or command line parameters? It is a bit odd to see the name of the device mapper pool in this script.

Reply:
yes agreed, I think BUILDKITE_BUILD_NUMBER may be more appropriate.

Reply:
I'd like to make it an explicit parameter rather than referencing environment variables. This script doesn't have to depend on BuildKite and environmental variables are hard to track since they are essentially global variables.

Reply:
In addition to, your PR description is great! Please put that in the commit message as well.

Reply:
sounds good, I can update the script to take an explicit parameter. In the pipeline, however, should we reference the BuildKite number to avoid concurrent builds overwriting the critest output file?